internal/exod/main.go

```diff
@@ -42,6 +44,20 @@ func Main(ctx context.Context) {
 	RunServer(ctx, cmd.Flags)
 }
 
+var makeHostHeaderMiddleware = func(validHosts []string) func(next http.Handler) http.Handler {
```

Comment:
This is a case where I think it would be much more Go-idiomatic to use a struct:

```go
type HostAllowListHandler struct {
  Hosts []string
  Next http.Handler
}

func (h *HostAllowListHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
  ...
}
```

The result is shallower nesting, easier debugging (structs are transparent when printed, closures are opaque), etc.

I might also move this to the httputil package just so that it's not cluttering up reading of exod/main.

Reply:
Yeah, fair, that's much less gross.